admin/tool/policy/classes/output/page_viewdoc.php

```diff
@@ -53,20 +53,25 @@ class page_viewdoc implements renderable, templatable {
     /** @var string $url Return URL. */
     protected $returnurl = null;
 
-    /** @var bool Can the user view draft versions. */
-    protected $canviewdraftrevision;
+    /** @var bool Can the user view all the versions or only their own or the current ones? */
+    protected $canviewallversions;
+
+    /** @var int $url User id who wants to view this page. */
+    protected $userid = null;
 
     /**
      * Construct this renderable.
      * @param int $policyid The policy id for this page.
-     * @param int $versionid The version id to show.
+     * @param int $versionid The version id to show. If none is specified is used the current version id.
      * @param string $returnurl Return URL.
+     * @param int $userid The userid which wants to view this policy version.
      */
-    public function __construct($policyid, $versionid = 0, $returnurl = null) {
+    public function __construct($policyid, $versionid = 0, $returnurl = null, $userid = 0) {
```

Comment:
What is the meaning of visiting a page like `view.php?userid=7` ?

It should always use just the current $USER imho. Why would you need to provide someone else's id? If the user has the capability to see acceptances or manage documents, she passes. If not, then look if the acceptance exists. If it does, the check passes. If not, then you are not allowed to see that policy. Am I missing something?

Reply:
I added this for the acceptbehalf functionality... The $USER of the parent/mentor doesn't correspond to the one who needs to view accepted documents (child/mentee) but he/she needs to have access to this policies (for instance to know which of them has been agreed by his/her child/mentee).

Reply:
I see. Thanks.